### ISSUE TITLE
flow fails to save protobuf file

### ISSUE THREAD

**Issue by bafu** [thtrieu/darkflow#239]
Case 1: FileNotFoundError exception ([log](https://github.com/thtrieu/darkflow/files/1016220/flow-savepb-tinyyolo_file-not-found-error.txt))

Executed

```
$ python3 flow --model cfg/tiny-yolo.cfg --load bin/tiny-yolo.weights --savepb 
```

and got the exception

```
...
Rebuild a constant version ...
Traceback (most recent call last):
  File "flow", line 64, in <module>
    tfnet.savepb(); exit('Done')
  File "/home/bafu/codes/darkflow/darkflow/net/build.py", line 183, in savepb
    os.makedirs(os.path.dirname(name), exist_ok=True)
  File "/usr/lib/python3.4/os.py", line 237, in makedirs
    mkdir(name, mode)
FileNotFoundError: [Errno 2] No such file or directory: ''
```
---
Case 2: Segmentation fault ([log](https://github.com/thtrieu/darkflow/files/1016219/flow-savepb-tinyyolo_segmentation-fault.txt))

Executed

```
$ python3 flow --model cfg/tiny-yolo.cfg --load bin/tiny-yolo.weights --savepb graph-tiny-yolo.pb
```

and got the exception

```
...
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 1
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 1:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:09:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:1) -> (device: 1, name: GeForce GTX TITAN X, pci bus id: 0000:05:00.0)
[1]    14107 segmentation fault (core dumped)  python3 flow --model cfg/tiny-yolo.cfg --load bin/tiny-yolo.weights --savepb
```

In `/var/log/syslog`

```
...
May 20 22:13:44 gc2 kernel: [3120037.057852] python3[14107]: segfault at 3038 ip 00007fb5b902a404 sp 00007ffda76de0c0 error 4 in libpthread-2.19.so[7fb5b9020000+19000]
```

---
Environment

* Darkflow: commit `3fe6b33`
* TensorFlow: 1.1.0
* [tiny-yolo](https://www.pjreddie.com/darknet/yolo/) (trained with COCO dataset, md5sum):
  * `b8f3314dbc8c37f9873798ec0b706d4e`  tiny-yolo.cfg
  * `dc9f663d36eb32cd2f9c158ffcd8ade0`  tiny-yolo.weights

**Comment by bafu** [comment]
In the environment above, inference test command also got segmentation fault error:

```
$ python3 flow --test test --model cfg/tiny-yolo.cfg --load bin/tiny-yolo.weights
...
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 1 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 1:   Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:09:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:1) -> (device: 1, name: GeForce GTX TITAN X, pci bus id: 0000:05:00.0)
Finished in 1.4301395416259766s

[1]    31895 segmentation fault (core dumped)  python3 flow --test test --model cfg/tiny-yolo.cfg --load
```

I suspect that the root cause is the same one, so I do not file a new bug.

**Comment by thtrieu** [comment]
This should be fixed in #240, please test again and tell me if it works.

**Comment by bafu** [comment]
@thtrieu This issue is fixed by pull request #240 (verified on commit `9de53ff`). Close this issue and thank for your and @abagshaw's efforts!
